pkg/utils/sets/sets.go

```diff
@@ -45,11 +47,28 @@ func NewComplementSet(values ...string) Set {
 	}
 }
 
+// Hash provides a hash function so we can generate a good hash for Set which has no public fields.
+func (s Set) Hash() (uint64, error) {
+	key := struct {
+		Values     sets.String
+		Complement bool
+	}{
+		s.values,
+		s.complement,
+	}
+	return hashstructure.Hash(key, hashstructure.FormatV2, nil)
+}
+
 // DeepCopy creates a deep copy of the set object
 // It is required by the Kubernetes CRDs code generation
 func (s Set) DeepCopy() Set {
```

Comment:
Remind me where we're deep copying the set?

Reply:
The generated DeepCopy method for Requirements calls Set's DeepCopy method.